pkg/pkgrepository/app_deploy.go

```diff
@@ -30,13 +30,105 @@ func (a *App) deploy(tplOutput string) exec.CmdRunResult {
 			return exec.NewCmdRunResultWithErr(fmt.Errorf("Preparing kapp: %s", err))
 		}
 
-		return kapp.Deploy(tplOutput, a.startFlushingAllStatusUpdates, func(exec.CmdRunResult) {})
+		// TODO: should we inject this somewhere else? - there's some other rules that might interact with it...
+		fmt.Println("sending to kapp deploy: \n", appendRebaseRule(tplOutput))
+		return kapp.Deploy(appendRebaseRule(tplOutput), a.startFlushingAllStatusUpdates, func(exec.CmdRunResult) {})
 
 	default:
 		return exec.NewCmdRunResultWithErr(fmt.Errorf("Unsupported way to deploy"))
 	}
 }
 
+func appendRebaseRule(tplOutput string) string {
+	return tplOutput + `
+---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+rebaseRules:
+- ytt:
+    overlayContractV1:
+      overlay.yml: |
+        #@ load("@ytt:data", "data")
+        #@ load("@ytt:json", "json")
+        #@ load("@ytt:overlay", "overlay")
+        #@ load("@ytt:struct", "struct")
+
+        #@ def get_rev(annotations):
+        #@  if hasattr(annotations, "packaging.carvel.dev/repo-rev"):
+        #@    return int(annotations["packaging.carvel.dev/repo-rev"])
+        #@  else:
+        #@    return -1
+        #@  end
+        #@ end
+        #@ if not hasattr(data.values.existing.metadata.annotations, "packaging.carvel.dev/package-repository-ref"):
+        #@   msg = "Error: cannot overwrite package " + data.values.existing.metadata.name + " because it was not created by a package repository."
+        #@   print(msg)
+        #@   fail(msg)
+        #@ end
+
+        #@ def annotations_are_identical(existing_anns, new_anns):
+        #@   carvel_anns = {"kapp.k14s.io/disable-original": True,
+        #@                   "kapp.k14s.io/disable-wait": True,
+        #@                   "packaging.carvel.dev/package-repository-ref": True,
+        #@                   "kapp.k14s.io/identity": True,
+        #@                   "kapp.k14s.io/create-strategy": True}
+        #@   return {x: v for x,v in existing_anns.items() if x not in carvel_anns} == {x: v for x,v in new_anns.items() if x not in carvel_anns}
```

Comment:
i think i'll write a `filter` func to reduce the `{x: v for x,v in .... }`

Reply:
yeh, some kind of filter or comparison function would probably make it more readable.